tests/config/test_document_type_config.py

```diff
@@ -160,3 +163,80 @@ def test_validate_document_invalid_extension(invalid_extension, valid_document):
         match=r"The extension provided for document type .* is not alphanumeric.",
     ):
         validate_document_type_config("ext", valid_document)
+
+
+@pytest.mark.skipif(sys.platform != "darwin", reason="Test runs only on macOS")
+def test_document_type_macOS_config_with_mimetype_single(valid_document):
+    """Valid document types don't raise an exception when validated.
+
+    application/pdf is the only valid MIME type for PDF files.
+    """
+    valid_document["mime_type"] = "application/pdf"
+    validate_document_type_config("ext", valid_document)
+    assert "LSItemContentTypes" in valid_document["macOS"].keys()
+    assert valid_document["macOS"]["LSItemContentTypes"] == "com.adobe.pdf"
+
+
+@pytest.mark.skipif(sys.platform != "darwin", reason="Test runs only on macOS")
+def test_document_type_macOS_config_with_mimetype_list(valid_document):
+    """Valid document types don't raise an exception when validated.
+
+    text/vcard is _not_ the only valid MIME type for vCard files, others are
+    text/directory and text/x-vcard so a list if MIME types is returned
+    internally but should still resolve to public.vcard
+    """
+    valid_document["mime_type"] = "text/vcard"
+    validate_document_type_config("ext", valid_document)
+    assert "LSItemContentTypes" in valid_document["macOS"].keys()
+    assert valid_document["macOS"]["LSItemContentTypes"] == "public.vcard"
+
+
+@pytest.mark.skipif(sys.platform != "darwin", reason="Test runs only on macOS")
+def test_document_type_macOS_config_with_unknown_mimetype(valid_document):
+    """Valid document types don't raise an exception when validated.
+
+    Here, a MIME type is provided that is not known to be valid for any file.
+    That means that LSItemContentTypes should _not_ be set.
+    """
+    valid_document["mime_type"] = "custom/mytype"
+    validate_document_type_config("ext", valid_document)
+    assert "LSItemContentTypes" not in valid_document["macOS"].keys()
+
+
+@pytest.mark.skipif(sys.platform != "darwin", reason="Test runs only on macOS")
+def test_mime_type_to_uti_with_nonexisting_coretypes_file(monkeypatch):
+    """Test that mime_type_to_UTI returns None if the coretypes file doesn't exist."""
+    monkeypatch.setattr(utils, "CORETYPES_PATH", "/does/not/exist")
+    assert utils.mime_type_to_UTI("application/pdf") is None
```

Comment:
For consistency, this should either (a) be test at the `validate_document_type_config()` level (i.e., a reproduction of the actual `application/pdf` case, but with the "couldn't find UTI data" response, or a separate test in the macOS module that does a couple of superficial direct checks of `mime_type_to_UTI()`.

Reply:
All the other tests in this file are based on invoking `validate_document_type_config()`; but this test is invoking `utils.mime_type_to_UTI`. The test suite is broadly organised by test function; a test of `mime_type_to_UTI` should either be in a test file for that method; or the entry point for for this specific test should be modified to use `validate_document_type_config()` (i.e., testing the same thing, but doing so by setting up test conditions that will result in performing a mime type lookup).

To that end - this test file should also be called `test_validate_document_type_config`.